### ISSUE TITLE
[Scaler Proposal] Azure IoT Hubs

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#214]
Proposal - Azure IoT Hubs would be a great scaler for IoT scenarios

**Comment by thdotnet** [comment]
Agreed

**Comment by jeffhollan** [comment]
IoT Hub and Event Hubs "consume" the same way today.  For example, with Azure Functions the IoT Hub trigger and Event Hub trigger are both really just 'event hub' trigger.  Would that also work here?  Is there any need to build something specific to Azure IoT Hubs, or would it be more or less an alias to Event Hubs?

**Comment by tomkerkhove** [comment]
That's correct but as you suggest I think we need to have an alias for this as a lot of customers are not aware of it (and frankly they shouldn't care).

So I'm fine with just an alias, unless there is another reasonable scenario in which people would want to scale.

**Comment by jeffhollan** [comment]
Awesome.  Let's see if we can add it as an alias and will call it out in the docs as such as well.  

https://github.com/kedacore/keda/issues/14

**Comment by tomkerkhove** [comment]
@jeffhollan would you mind adding it to the Scaler Prio wiki please?
Thanks!

**Comment by jonmikeli** [comment]
I have just seen that the feature has been removed from Planned (Committed).
Any specific issue? or just about priorities/capacity? (if so, I cannot promise anything in the short term but I can maybe bring my small and humble contribution during the summer).

**Comment by tomkerkhove** [comment]
No particular reason other than moving from GitHub Projects to GitHub Projects (preview) :)

Always happy to hear about customer scenarios to help prior or receive contributions though.

**Comment by SushmithaVReddy** [comment]
@tomkerkhove @jeffhollan Adding to the alias discussion - I've observed there is a difference in connection strings obtained for the Event hub and IoT hub through portal and the event hub scalar code doesn't seem to handle the format used in IoT hub.

EvenHub connection string - Endpoint=sb://<NamespaceName>.servicebus.windows.net/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>
IoT hub connection string - HostName=<HubName>.azure-devices.net;<KeyName>;SharedAccessKey=<KeyValue>

Having said that, do we have an azure sdk that can be used if we create a new IoT scalar?  I'm doubtful about leveraging https://github.com/Azure/azure-sdk-for-go for IoT Hub.

**Comment by tomkerkhove** [comment]
Then we are out of luck I think - Either we go with an external scaler similar to the Cosmos DB Changefeed or we put it in the fridge until it's there.

**Comment by SushmithaVReddy** [comment]
> Then we are out of luck I think - Either we go with an external scaler similar to the Cosmos DB Changefeed or we put it in the fridge until it's there.

@raorugan any thoughts?

**Comment by raorugan** [comment]
I think it will be good to get the roadmap of IOT hub if they have plans to enable Azure SDK for go lang

**Comment by tomkerkhove** [comment]
Feel free to open an issue there and mention this issue so that they are linked, but for now we can't add it to KEDA core then.

**Comment by AiHaibara** [comment]
> @tomkerkhove @jeffhollan Adding to the alias discussion - I've observed there is a difference in connection strings obtained for the Event hub and IoT hub through portal and the event hub scalar code doesn't seem to handle the format used in IoT hub.
> 
> EvenHub connection string - Endpoint=sb://.servicebus.windows.net/;SharedAccessKeyName=;SharedAccessKey= IoT hub connection string - HostName=.azure-devices.net;;SharedAccessKey=
> 
> Having said that, do we have an azure sdk that can be used if we create a new IoT scalar? I'm doubtful about leveraging https://github.com/Azure/azure-sdk-for-go for IoT Hub.

@SushmithaVReddy 
there is a eventhub built-in endpoints in Hub settings in IotHub.
it is the same format with the eventhub connection string.
but if i add identity role permission to iothub not working for access to it's built-in eventhub.
